modules/account-lib/src/coin/hbar/coinTransferBuilder.ts

```diff
@@ -0,0 +1,144 @@
+  constructor(message: string);
+}
+
+export class CoinTransferBuilder extends TransferBuilder {
+  // TODO: [BG-51282] Deprecate once wp work for multi recipients
+  private _toAddress: string;
+  private _amount: string;
+
+  constructor(_coinConfig: Readonly<CoinConfig>) {
+    super(_coinConfig);
+  }
+
+  /** @inheritdoc */
+  protected async buildImplementation(): Promise<Transaction> {
+    this._txBodyData.transfers = this.buildTransferData();
+    return await super.buildImplementation();
+  }
+
+  private buildTransferData(): proto.ITransferList {
+    let totalSend = new BigNumber(0);
+    const accountAmounts: proto.AccountAmount[] = [
+      {
+        accountID: this.buildAccountData(this._source.address),
+        amount: Long.fromInt(0),
+      }, // sender
+    ];
+    // add recipients and update send amount
+    this._recipients.map((recipient) => {
+      accountAmounts.push(
+        { accountID: this.buildAccountData(recipient.address), amount: Long.fromString(recipient.amount) }, // recipient
+      );
+      totalSend = totalSend.plus(recipient.amount);
+    });
+    accountAmounts[0].amount = Long.fromString(totalSend.toString()).negate(); // update sender send amount
+
+    return {
+      accountAmounts: accountAmounts,
+    };
+  }
+
+  /** @inheritdoc */
+  initBuilder(tx: Transaction): void {
+    super.initBuilder(tx);
+    const transferData = tx.txBody.cryptoTransfer;
+    if (transferData && transferData.transfers && transferData.transfers.accountAmounts) {
+      this.initTransfers(transferData.transfers.accountAmounts);
+    }
+  }
+
+  /**
+   * Initialize the transfer specific data, getting the recipient account
+   * represented by the element with a positive amount on the transfer element.
+   * The negative amount represents the source account, so it's ignored.
+   *
+   * @param {proto} transfers - Array of objects which contains accountID and transferred amount
+   */
+  protected initTransfers(transfers: proto.IAccountAmount[]): void {
+    transfers.forEach((transferData) => {
+      const amount = Long.fromValue(transferData.amount!);
+      if (amount.isPositive()) {
+        this.send({
+          address: stringifyAccountId(transferData.accountID!),
+          amount: amount.toString(),
+        });
+      }
+    });
+  }
+
+  // region Transfer fields
+  /** TODO: [BG-51282] Deprecate to and amount once wp has been fixed
+      Currently work for one recipient by using exatcly one of to + amount or send function
+   */
+  /**
+   * Set the destination address where the funds will be sent,
+   * it may take the format `'<shard>.<realm>.<account>'` or `'<account>'`
+   *
+   * @param {string} address - The address to transfer funds to
+   * @returns {TransferBuilder} - The builder with the new parameter set
+   */
+  to(address: string): this {
+    if (this._recipients.length > 0) {
+      throw new DuplicateMethodError('Invalid method: send already used');
+    }
```

Comment:
it seems odd to make builders immutable. what was the rationale behind this?

I think it's acceptable to be able to set / re-set properties without creating a brand new builder.

Also, you might want to re-construct a builder from a serialized transaction and update certain properties.

Reply:
This is just temporary so people don't use both method since only recipient 0 is currently the one that will be processed in wp, will remove this one wp updated for multi-recipients.

Reply:
Actually right, account-lib is independent from wp, so I can make to and amount used normally for now since it's not necessarily be processed in wallet-platform anyway

Reply:
ok, thanks for explaining